### DISCUSSION TITLE
Separate background when sun inclination is low

### DISCUSSION THREAD

**Discussion by opusforlife2** [breezy-weather/breezy-weather#1039] [Ideas]
### Describe your idea

The app currently goes from bright orange sunny/bright sky blue to a suddenly starry/non-starry night. Depending on the season, there can be long evening periods where the sun is too oblique to see, and yet the sky remains bright enough that it can't be called night by any stretch of the imagination.

A separate background for evenings would be great, maybe something in muted blue and purple hues?

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.3 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
The definition of "evening" is not the same in every country, depending on your location timezone vs solar timezone, depending if you live near poles where sun can be there all day... or the opposite the sun doesn't rise.

The day/night segmentation is not based on defined periods of the day, but on sunrise/sunset which makes it reliable everywhere in the world.

That means, as you noticed, that the weather condition will be "Sunny" even if the sun is about to set.

**Reply by opusforlife2** [reply]
Oh. Is the background merely an indicator of time, then? Is the app not meant to follow weather outside? I'd expect summer inside the Arctic Circle to not show sunsets at all in the app.

**Reply by papjul** [reply]
> Is the background merely an indicator of time, then?

It's the opposite, it's not based on time but sun:

> The day/night segmentation is not based on defined periods of the day, but on sunrise/sunset which makes it reliable everywhere in the world.

This sentence implied that if sunrise and sunset is undefined (polar night, no sun from all day), you will only get night backgrounds

**Reply by opusforlife2** [reply]
So there's just 2 periods, between sun-up time and sun-down time, instead of roughly 4 periods of a day that vary by season and location - morning, afternoon, evening, night.

**Reply by papjul** [reply]
There has never been 4 periods in Breezy Weather.

I'm no longer sure what you're talking about.

If you're talking about daily charts **data**, we split the day in 12-hour periods, as documented here: https://github.com/breezy-weather/breezy-weather/blob/main/docs/HOMEPAGE.md#daily-forecast

If you're talking about the background, we just show a lighter background when the sun is up, and a darker background when the sun is down. This has nothing to do with any period.

Also, the concept of season is mostly for temperate and polar regions. What does "morning, afternoon, evening, night" when you live near the Arctic? I would say that for them, the morning is still the morning even if it's "night" outside at some periods of the day.

TL;DR: there are no plans to make the background varies depending on the inclination of the sun that would be different depending on the longitude/latitude of the location. That's just too much for a an almost inexistent gain. If you know any open source app that does that, feel free to share, so I can still take a look.

**Reply by opusforlife2** [reply]
I was initially thinking of something far less complicated. Services provide city-based sunrise and sunset times (I assume, since the app lets you select a city). So if the time between sunrise and sunset is, say, 15 hours, that could be divided (at the simplest) into three 5 hour periods.

The last period could have its own evening animation. And the first period could have a morning animation, come to think about it. The middle period would continue to have the current orange sun animation. Sunset to sunrise wouldn't need to change either.

**Reply by papjul** [reply]
When the sun is up for 15 hours, the sun is not setting after 10 hours.

When the sun is up for 3 hours, the sun is not up in the air, it's always low.

So it's not that simple, it's really the inclination that defines the luminosity.

**Reply by opusforlife2** [reply]
Doesn't the [sunrise equation](https://en.wikipedia.org/wiki/Sunrise_equation) approximate this?

**Reply by papjul** [reply]
I guess you could base your work on it. And you plan on designing 2 additional backgrounds for each weather condition? Or probably just one since sunrise and sunset should be the same in terms of light?

**Reply by opusforlife2** [reply]
I'm just requesting the feature. I don't know nearly enough to propose a PR. But it at least seems feasible after this discussion.

**Reply by papjul** [reply]
It's not because it's feasible that it has to be done. Adding overhead is not always a good thing. I like the way it is, and no one has design/technical plans, so I guess it can be closed

**Reply by opusforlife2** [reply]
Are immediately available designs or technical plans your threshold for feature requests to remain open? It's an idea. An external contributor could come across this and find a way to do it efficiently, making way for a PR. A closed discussion will just be buried along with the rest.

**Reply by papjul** [reply]
Not necessarily.

It's very unlikely someone will come just to implement this, because it requires a high effort for a little gain. Currently, we are based on a basic boolean `isDaylight`. We would need to re-implement a lot of things to switch to (probably) an `enum`. But it's actually more complex than that since we need to calculate the equation (which is an unnecessary overhead IMO).

And we don't even have a design of what each weather condition would look like to see if that's really worth it. I can see an improvement for the sunny condition, but what about the others? If it's just for one condition, it's not worth adding such complexity.

If you were planning on implementing it, we could have discussed how you planned to work your way.

Otherwise, if it's up to the project, we will decide it doesn't fit for the reasons mentioned, and we won't open it to contributions for the same reasons.

I hope to implement ability to extend the app in the short term (as separate projects), so if someone is really motivated later on, that will be the way they will have to do it.

**Reply by opusforlife2** [reply]
Oh. I see. The gulf in complexity between the current state and the intended one is too vast.

Alright, thanks for the discussion! TIL sunrise equation is a thing.
